lib/smack/SmackRep.cpp

```diff
@@ -758,36 +935,99 @@ const Expr* SmackRep::declareIsExternal(const Expr* e) {
 }
 
 string SmackRep::getPtrType() {
```

Comment:
This method should always return `ref`.

In fact, we don't really even need this method anymore, and all of those messy string concatenations with calls to `getPtrType()` could be eliminated.